frontend/src/features/admin-form/create/workflow/components/WorkflowContent/EditStepBlock/RespondentBlock.tsx

```diff
@@ -170,19 +204,426 @@ const DynamicRespondentOption = ({
+      type: 'text/csv',
+    } as File
+  }, [standardCsvDownloadFileName])
+
+  useEffect(() => {
+    if (selectedConditionalFieldOptionsToRecipientsMap) {
+      setCsvFile(placeholderOptionToEmailMappingCsv)
+    } else {
+      setCsvFile(null)
+    }
+  }, [
+    placeholderOptionToEmailMappingCsv,
+    setCsvFile,
+    selectedConditionalFieldOptionsToRecipientsMap,
+  ])
+
+  const { isOpen, onOpen, onClose } = useDisclosure()
+
+  const handleCsvDownload = () => {
+    if (!selectedConditionalFieldOptionsToRecipientsMap) return
+    const csvData = {
+      fields: ['Options', 'Add email(s) in this column'],
+      data: Object.entries(selectedConditionalFieldOptionsToRecipientsMap).map(
+        ([option, recipients]) => [option, recipients.join(',')],
+      ),
+    }
+
+    const csvString = Papa.unparse(csvData, {
+      header: true,
+      delimiter: ',',
+      skipEmptyLines: 'greedy',
+    })
+    const csvBlob = new Blob([csvString], {
+      type: 'text/csv',
+    })
+    const csvFile = new File([csvBlob], standardCsvDownloadFileName, {
+      type: 'text/csv',
+    })
+    downloadFile(csvFile)
+  }
+
+  const handleSkeletonCsvDownload =
+    (formId: string = '') =>
+    () => {
+      const getFieldOptions = (conditionalFieldId: string) => {
+        const conditionalField = conditionalFormFields.find(
+          (field) => field._id === conditionalFieldId,
+        )
+        return conditionalField?.fieldOptions
+      }
+      const generateCsvContent = (fieldOptions: string[] | undefined) => {
+        const headerRow = ['Options', 'Add email(s) in this column']
+        const optionsRows = fieldOptions?.map((field) => [field, '']) ?? []
+        const jsonContent = [headerRow, ...optionsRows]
+        return Papa.unparse(jsonContent, {
+          header: true,
+          delimiter: ',',
+        })
+      }
+
+      const csvStringToFile = (csvString: string, downloadFileName: string) => {
+        const csvBlob = new Blob([csvString], {
+          type: 'text/csv',
+        })
+        const csvFile = new File([csvBlob], downloadFileName, {
+          type: 'text/csv',
+        })
+        return csvFile
+      }
+
+      const conditionalFieldId = watch('conditional_field')
+      if (!conditionalFieldId) return
+
+      const fieldOptions = getFieldOptions(conditionalFieldId)
+      const csvContent = generateCsvContent(fieldOptions)
+      const csvFile = csvStringToFile(
+        csvContent,
+        `conditional_routing_form_${formId}_field_${selectedConditionalFieldTitle}_mapping.csv`,
+      )
+      downloadFile(csvFile)
+    }
+
+  const { editOptionToRecipientsMutation } = useEditFormField()
+
+  const handleConditionalRoutingConfigSubmit =
+    (conditionalFieldId: string | undefined) =>
+    async (data: ConditionalRoutingConfig) => {
+      if (!(data.csvFile && conditionalFieldId)) {
+        return
+      }
+
+      const conditionalRoutingCsvString = await parseCsvFileToCsvString(
```

Comment:
Spotting a potential reusable utils function here. Likely that this parser and the one below in `validateCsvFile()` is to be kept consistent. Would recommend to create a local utils to maintain consistency.

Reply:
fixed in [8ddc9a7](https://github.com/opengovsg/FormSG/pull/7804/commits/8ddc9a758ff8a6e7a0d28523b1b4153d06fdcb04). stored in the same file since it is specific to csv template and only used in `ConditionalRoutingOption`